cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/azureDeployment.tsx

```diff
@@ -36,7 +37,7 @@ const useAzureDeployment = () => {
             ceipOptIn: formInfo[AZURE_FIELDS.CEIP_OPT_IN],
             clusterName: formInfo[AZURE_FIELDS.CLUSTER_NAME],
             controlPlaneFlavor: formInfo[AZURE_FIELDS.CONTROL_PLANE_FLAVOR],
-            controlPlaneMachineType: retrieveAzureInstanceType(formInfo[AZURE_FIELDS.NODE_PROFILE]),
+            controlPlaneMachineType: nodeTypes[formInfo[AZURE_FIELDS.NODE_PROFILE]],
```

Comment:
Does this mean that nodeTypes (as stored in the RESOURCES section) only contains a single node instance type (the default, presumably) segmented by NODE_PROFILE?
I think we may want to put our heads together, because for AWS what we're doing is having the RESOURCE section have ALL the node types (segmented by AZ) and then when we choose one (the default) we then store the chosen one in the FORM section.
It would probably be good to decide on a common approach.

Reply:
@swalner-vmware the nodeTypes is a map which contains a node type  for each node profile: 
"nodeType":{"SINGLE_NODE":"Standard_B2ms","HIGH_AVAILABILITY":"Standard_B2ms","PRODUCTION_READY":"Standard_B4ms"}

Reply:
rewrote the logic.